### ISSUE TITLE
Bug report: formatters on `version` cause incorrect output in `cask info`

### ISSUE THREAD

**Issue by Numbermaniac** [caskroom/homebrew-cask#27660]
#### Description of issue

If an Artifact for a cask interpolates `version`, it displays correctly in `cask info`. However, if it uses a formatter, in this case `.before_comma`, it does not seem to do the format in `cask info`.

<details><summary>Output of `brew cask info boinc --verbose` (Artifact prints correctly)</summary>

```
boinc: 7.6.33
https://boinc.berkeley.edu/
Not installed
From: https://github.com/caskroom/homebrew-cask/blob/master/Casks/boinc.rb
==> Names
Berkeley Open Infrastructure for Network Computing
BOINC
==> Artifacts
boinc_7.6.33_macOSX_x86_64/BOINC Installer.app/Contents/Resources/BOINC.pkg (pkg)

```
</details>

<details><summary>Output of `brew cask info pycharm-eap --verbose` (Artifact prints incorrectly)</summary>

```
pycharm-eap: 2016.3,163.6512.5
https://confluence.jetbrains.com/display/PYH/JetBrains+PyCharm+Preview+(EAP)
Not installed
From: https://github.com/caskroom/homebrew-versions/blob/master/Casks/pycharm-eap.rb
==> Name
PyCharm EAP
==> Artifacts
PyCharm #{version.before_comma} EAP.app (app)

```
</details>
<details><summary>Output of `brew doctor`</summary>

```
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry and just ignore them. Thanks!

Warning: "config" scripts exist outside your system or Homebrew directories.
`./configure` scripts often look for *-config scripts to determine if
software packages are installed, and what additional flags to use when
compiling and linking.

Having additional scripts in your path can confuse software installed via
Homebrew if the config script overrides a system or Homebrew provided
script of the same name. We found the following "config" scripts:
  /Applications/Wine Devel.app/Contents/Resources/wine/bin/xml2-config
  /Applications/Wine Devel.app/Contents/Resources/wine/bin/xslt-config


Warning: Some installed formula are missing dependencies.
You should `brew install` the missing dependencies:
  brew install gdbm sqlite

Run `brew missing` for more details.
```
</details>

<details><summary>Output of `brew cask doctor`</summary>

```
==> macOS Release:
10.12.1
==> Hardware Architecture:
intel-64
==> Ruby Version:
2.0.0-p648
==> Ruby Path:
/System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/bin/ruby
==> Homebrew Version:
Homebrew 1.1.2-75-gebe4549
Homebrew/homebrew-core (git revision 0b4c9; last commit 2016-12-11)
==> Homebrew Executable Path:
/usr/local/bin/brew
==> Homebrew Cellar Path:
/usr/local/Cellar
==> Homebrew Repository Path:
/usr/local/Homebrew
==> Homebrew Origin:
https://github.com/Homebrew/brew
==> Homebrew-Cask Version:
Homebrew-Cask 1.1.2-75-gebe4549
caskroom/homebrew-cask (git revision a76661; last commit 2016-12-12)
==> Homebrew-Cask Install Location:
<NONE>
==> Homebrew-Cask Staging Location:
/usr/local/Caskroom
==> Homebrew-Cask Cached Downloads:
~/Library/Caches/Homebrew/Cask
0 files, 0B
==> Homebrew-Cask Default Tap Path:
/usr/local/Homebrew/Library/Taps/caskroom/homebrew-cask
==> Homebrew-Cask Alternate Cask Taps:
/usr/local/Homebrew/Library/Taps/caskroom/homebrew-versions
==> Homebrew-Cask Default Tap Cask Count:
3479
==> Contents of $LOAD_PATH:
/usr/local/Homebrew/Library/Homebrew/cask/lib
/usr/local/Homebrew/Library/Homebrew
/Library/Ruby/Gems/2.0.0/gems/json-2.0.2/lib
/Library/Ruby/Gems/2.0.0/extensions/universal-darwin-16/2.0.0/json-2.0.2
/Library/Ruby/Gems/2.0.0/gems/psych-2.1.1/lib
/Library/Ruby/Gems/2.0.0/extensions/universal-darwin-16/2.0.0/psych-2.1.1
/Library/Ruby/Site/2.0.0
/Library/Ruby/Site/2.0.0/x86_64-darwin16
/Library/Ruby/Site/2.0.0/universal-darwin16
/Library/Ruby/Site
/System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/vendor_ruby/2.0.0
/System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/vendor_ruby/2.0.0/x86_64-darwin16
/System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/vendor_ruby/2.0.0/universal-darwin16
/System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/vendor_ruby
/System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0
/System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/x86_64-darwin16
/System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/universal-darwin16
==> Contents of $RUBYLIB Environment Variable:
<NONE>
==> Contents of $RUBYOPT Environment Variable:
<NONE>
==> Contents of $RUBYPATH Environment Variable:
<NONE>
==> Contents of $RBENV_VERSION Environment Variable:
<NONE>
==> Contents of $CHRUBY_VERSION Environment Variable:
<NONE>
==> Contents of $GEM_HOME Environment Variable:
<NONE>
==> Contents of $GEM_PATH Environment Variable:
<NONE>
==> Contents of $BUNDLE_PATH Environment Variable:
<NONE>
==> Contents of $PATH Environment Variable:
PATH="/Applications/Wine Devel.app/Contents/Resources/start/bin:/Applications/Wine Devel.app/Contents/Resources/wine/bin:/Library/Frameworks/Python.framework/Versions/2.7/bin:/usr/local/sbin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/opt/X11/bin:/usr/local/Homebrew/Library/Taps/buo/homebrew-cask-upgrade/cmd:/usr/local/Homebrew/Library/Homebrew/cask/cmd:/usr/local/Homebrew/Library/Homebrew/shims/scm"
==> Contents of $SHELL Environment Variable:
SHELL="/bin/bash"
==> Contents of Locale Environment Variables:
LANG="en_AU.UTF-8"
```
</details>

**Comment by miccal** [comment]
This is because the `app` stanza in [pycharm-eap](https://github.com/caskroom/homebrew-versions/blob/master/Casks/pycharm-eap.rb) is missing the double quotes required for the interpolation to work.

**Comment by miccal** [comment]
Fixed in [#2994](https://github.com/caskroom/homebrew-versions/pull/2994) - thanks for reporting the issue.
